### ISSUE TITLE
Revise communicator specification in Fortran interface to itsolv

### ISSUE THREAD

**Issue by pjknowles** [molpro/iterative-solver#101]
In GitLab by @KnowlesPJ on Dec 29, 2020, 07:58

At the moment, it's difficult to understand from a user point of view. It's not very clear what to do with the `lmppx` and `pcomm` parameters. Is `pcomm` just for the profiler, or for the whole thing?

There is also the problem that we are trying to stay aloof from MPI fortran, so do not want to include `mpif.h` at the fortran level. However we can be sure that at the C level, `MPI_Comm MPI_Comm_f2c(MPI_Fint comm)` is available. This means that when calling from Fortran, we can pass a communicator as an integer, provided the size of that integer is at least as big as `MPI_Fint`. One suggestion is to make the interface for a communicator parameter `int64` from `iso_fortran_env`; this will always work. Probably we would want to abstract this integer kind with its own name in our module.  This provides enough to support a client program that uses fortran MPI - the communicator can be passed successfully.

This then leaves the problem of how to specify from fortran the special communicators, which include `MPI_COMM_WORLD`, `MPI_COMM_SELF`, ppidd's world communicator, without using Fortran MPI.  This can be achieved by providing some functions in our module to return these - world and self should be enough. There should be setter functions for "world" so that a particular environment can be set up by the client.

All of this can be implemented in the non-mpi case because there's now sufficient abstraction to make it possible.

In summary it is suggested that all of the initialise functions have a single int64 communicator parameter that is optional. The default should be "world" as we maintain it in the module. The user can instead pass a call to the "self" function if desired. If the library is built without MPI, it doesn't matter - world and self are the same thing.
